requirements.txt

```diff
@@ -14,3 +14,4 @@ sortedcontainers~=2.0
 scipy
 sympy
 typing_extensions
+tqdm
```

Comment:
New dependency alert! 

In lieu of keeping cirq light, I'm hesitant to add tqdm as a dependency.

1. We should check if this is going to break g3
2. Can we make this somehow optional so that we reduce the dependency requirements for Cirq? (e.g. `cirq[ui]` or `cirq[visualization]` or via making the default value progress bar NoProgress) 
3. do we need to pin this? 
4. can we abstract it out better in our code instead of directly depending on it? That would help with testing progress logic as well as well as would

Reply:
If we're going to package our experiments/algorithms code in Cirq, we need to figure this out. For now, I can change it to default to no-progress but I think this hurts usability of this application code

Reply:
Fair point about if we package this together we'll have to deal with this - and that is the plan. 
I'll make a mental note of this. I checked 1.) and we should be fine (though I haven't run tests on that version).